### PR TITLE
move the header parts in a taglib

### PR DESCRIPTION
**Pull request by mawinter69**
the taglib allows plugins to make use of the searchbox and the login stuff while easily being able to modify all the other aspects of the header.

### Testing done
Manually verified that the header is not modified. Also in the downstream PR using the taglib works without problems.

see https://github.com/jenkinsci/customizable-header-plugin/pull/101

### Proposed changelog entries

- Provide header parts as a tag library.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/resources/jenkins/views/JenkinsHeader/headerContent.jelly
Comment (daniel-beck) [inline core/src/main/resources/jenkins/views/JenkinsHeader/headerContent.jelly:6]: This is now public API. Are we confident a redesign doesn't just randomly change this anytime soon?

Comment (mawinter69) [reply core/src/main/resources/jenkins/views/JenkinsHeader/headerContent.jelly:6]: Any substantial change to the header would require a change of the header version in the [PartialHeader](https://github.com/jenkinsci/jenkins/blob/b89863cf1f1ca7a27626774bd713f9a1d6fde2f6/core/src/main/java/jenkins/views/PartialHeader.java#L28C30-L28C57) so that plugins need to react. I might need to stick to the PartialHeader in the PR of my plugin (which is so far the only implementation of a header) https://github.com/jenkinsci/customizable-header-plugin/blob/295683015391e800be41c568b42ac80f6fd7c61b/src/main/java/io/jenkins/plugins/customizable_header/headers/LogoHeader.java#L9

**Review by timja — APPROVED** [review]
Makes sense, downstream changes look good

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
